haystack/components/preprocessors/recursive_chunker.py

```diff
@@ -0,0 +1,212 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import re
+from typing import Any, Dict, List
+
+from haystack import Document, component, default_from_dict, default_to_dict, logging
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class RecursiveChunker:
```

Comment:
RecursiveDocumentSplitter as a component name would be more similar to the existing DocumentSplitter/NLTKDocumentSplitter